nautobot/project-static/js/table_sorting_indicator.js

```diff
@@ -0,0 +1,46 @@
+document.addEventListener("DOMContentLoaded", function() {
+    // Select all header elements with the class "orderable" containing links
+    const headers = document.querySelectorAll(".orderable a");
+
+    headers.forEach(header => {
+        // Get the column name from the data attribute
+        const columnName = header.dataset.columnName;
+
+        // Add a click event listener to each header
+        header.addEventListener("click", function(event) {
+            // Prevent the default link behavior
+            event.preventDefault();
+
+            // Get the current URL and its search parameters
+            const currentUrl = new URL(window.location.href);
+            const searchParams = new URLSearchParams(currentUrl.search);
+            let sortParam = searchParams.get("sort");
+
+            if (sortParam === columnName) {
+                // If the column is already sorted in ascending order, switch to descending
+                searchParams.set("sort", "-" + columnName);
+            } else if (sortParam === "-" + columnName) {
+                // If the column is already sorted in descending order, reset to default
+                searchParams.delete("sort");
+                window.location.href = window.location.pathname; // Reload the page with the default view
```

Comment:
Do we want this to be added to the browser history or do we want to use a `location.replace` here? A client-side UI wouldn't require a page load to change the table order so the behavior would match `location.replace`, but our current UI behavior is to initiate a page load.

Reply:
very interesting, thank you! did not realize that, I can see how this would be cumbersome specially with the back button after several sortings.